lib/jwt/verify.rb

```diff
@@ -53,9 +53,14 @@ def verify_iss
 
       iss = @payload['iss']
 
-      return if Array(options_iss).map(&:to_s).include?(iss.to_s)
+      options_iss = Array(options_iss).map { |item| item.is_a?(Symbol) ? item.to_s : item }
 
-      raise(JWT::InvalidIssuerError, "Invalid issuer. Expected #{options_iss}, received #{iss || '<none>'}")
+      case iss
+      when *options_iss
```

Comment:
This is a pretty neat way to handle all the different possible values. Thanks for teaching me :)

Reply:
This comment can be ignored, feel that I might be overthinking this. Just need to write the behaviour down to be clear:)

Something like this will not work, because we stop calling the lambdas after the first truthy value.

The idea that we require the issuer but some are not allowed.
```
JWT.decode token, hmac_secret, true,
           iss: [->(issuer) { !issuer.blank? },
                 ->(issuer) { issuer != 'The issuer we do not accept' }],
           verify_iss: true,
           algorithm: 'HS256'
```

Reply:
Also supporting EOL rubies is biting this wonderful approach of using a `case`. It does not seem to work with methods

Reply:
Was taking to a few colleagues about this neat way of using a use-case and they pointed out that this works because the `#call` method is aliased to `#===` for procs. So maybe it would be more readable and less magical if the check would just be:

```
 return if options_iss.any? { |option| options === iss }
```

Reply:
What goes for the EOL rubies and the failing test related to passing a method, we could just not run the test with the older rubies until we officially drop the support for the older rubies.

Reply:
> Something like this will not work, because we stop calling the lambdas after the first truthy value.

My understanding is that the list is an "allowlist" (it's the case for strings) so there is no expectation to check further after one positive match.

> the `#call` method is aliased to `#===` for procs

Rubocop complains about any explicit use of `===`. Indeed this PR relies on the behavior of `===` for procs and regular expressions, in addition to the fact that `===` is the same as `==` for string instances. But I tend to agree with BBatsov on this, `===` is an internal tool to implement case matching and pattern matching, which are the user-facing language features to be used. It's like having to use `h = my_object.deconstruct_keys(:a, :b); a, b = h.values_at(:a, :b)` instead of `h => {a: , b:}`.

Reply:
> What goes for the EOL rubies and the failing test related to passing a method

I will try to make them pass.

Reply:
> I will try to make them pass.

No way - just skipped them

Reply:
You make a fair point, let's have it like this. I still think calling the explicit `#===` would be less code and a little more readable.